src/components/WatchlistWeeklyReport/WatchlistWeeklyReport.js

```diff
@@ -0,0 +1,162 @@
+import React, { useState } from 'react'
+import { connect } from 'react-redux'
+import cx from 'classnames'
+import Dialog from '@santiment-network/ui/Dialog'
+import Button from '@santiment-network/ui/Button'
+import Toggle from '@santiment-network/ui/Toggle'
+import Notification from '@santiment-network/ui/Notification'
+import { InputWithIcon } from '@santiment-network/ui/Input'
+import EmailImage from './EmailImage'
+import EmailSetting from '../../pages/Account/EmailSetting'
+import { showNotification } from '../../actions/rootActions'
+import { WATCHLIST_TOGGLE_MONITORING } from '../../actions/types'
+import styles from './WatchlistWeeklyReport.module.scss'
+
+const NOTIFICATION = {
+  connected: {
+    title: 'You have email address connected',
+    description:
+      'You’ll receive reports to the email address connected with your account',
+    className: styles.notification,
+    variant: 'info',
+    classes: {
+      title: styles.notificationTitle,
+      conent: styles.notificationDescription
+    }
+  },
+  notConnected: {
+    title: 'Please connect your email address',
+    description:
+      'We’ve noticed that you’re logged in with Metamask, please enter an email address which will be connected to your account',
+    className: cx(styles.notification, styles.notificationWarning),
+    variant: 'warning',
+    classes: {
+      title: styles.notificationTitle,
+      content: styles.notificationDescription
+    }
+  }
+}
+
+const STATUSES = {
+  loading: 'Email is verifying',
+  success: 'Email is correct',
+  error: 'Error during typing email'
+}
+
+const WatchlistWeeklyReport = ({
+  trigger,
+  isMonitored: initialIsMonitored,
+  email,
+  id,
+  name,
+  dispatchIsMonitored,
+  setNotification
+}) => {
+  const [isShown, setIsShown] = useState(false)
+  const [isMonitored, toggleIsMonitored] = useState(initialIsMonitored)
+  const [emailStatus, toggleEmailStatus] = useState()
+
+  const isEmailConnected = !!email
+
+  const close = () => {
+    setIsShown(false)
+    toggleEmailStatus(null)
+  }
+
+  const open = () => {
+    setIsShown(true)
+    toggleIsMonitored(initialIsMonitored)
+  }
+
+  const onSave = () => {
+    if (isEmailConnected && initialIsMonitored !== isMonitored) {
+      dispatchIsMonitored({ id, isMonitored })
+    }
+
+    if (!isEmailConnected) {
+      if (emailStatus === STATUSES.success) {
```

Comment:
`!isEmailConnected` and `emailStatus === STATUSES.success` could be written in a single `if` statement as `condition_a && condition_b`